src/molecules/Modal/Modal.native.js

```diff
@@ -0,0 +1,157 @@
+            onBackButtonPress={onClose}
+            backdropColor={theme.colors.shade[970]}
+          >
+            <ModalContentContainer type="centered">
+              {onClose ? (
+                <Flex alignSelf="flex-end">
+                  <Space padding={[0.5, 1, 0, 0]}>
+                    <View>
+                      <Button
+                        testID="close-button"
+                        variant="tertiary"
+                        variantColor="tone"
+                        icon="close"
+                        size="medium"
+                        onClick={onClose}
+                      />
+                    </View>
+                  </Space>
+                </Flex>
+              ) : null}
+              {React.Children.map(children, (child) => {
+                return React.cloneElement(child, { ...child.props, type });
+              })}
+            </ModalContentContainer>
+          </RNModal>
+        </Space>
+      );
+    }
+    if (type === 'fullscreen') {
+      return (
+        <Space margin={[`${getStatusBarHeight(true)}px`, 0, 0, 0]}>
+          <RNModal isVisible={visible} hasBackdrop={false} onBackButtonPress={onClose}>
+            <Flex flex={1}>
+              <ModalContentContainer type="fullscreen">
+                {React.Children.map(children, (child) => {
+                  return React.cloneElement(child, { ...child.props, type, onClose });
+                })}
+              </ModalContentContainer>
+            </Flex>
+          </RNModal>
+        </Space>
+      );
+    }
+    if (type === 'bottomsheet') {
+      return (
+        <Space margin={[0]}>
+          <Flex justifyContent="flex-end">
+            <RNModal
+              isVisible={visible}
+              swipeDirection={['down']}
+              onBackdropPress={onBackdropClick}
+              onSwipeComplete={onClose}
+              onBackButtonPress={onClose}
+              avoidKeyboard={true}
+              backdropColor={theme.colors.background[200]}
+            >
+              <ModalContentContainer type="bottomsheet">
+                <Flex alignItems="center">
+                  <Space padding={[1, 0, 1.5, 0]}>
+                    <View>
+                      <Size height={0.5} width={8}>
+                        <BottomSheetDragBar />
+                      </Size>
+                    </View>
+                  </Space>
+                </Flex>
+                {React.Children.map(children, (child) => {
+                  return React.cloneElement(child, { ...child.props, type });
+                })}
+              </ModalContentContainer>
+            </RNModal>
+          </Flex>
+        </Space>
+      );
+    }
+    return null;
+  };
+
+  return renderModal();
+};
+
+Modal.Header = ModalHeader;
+Modal.Content = ModalContent;
+Modal.Footer = ModalFooter;
+
+Modal.propTypes = {
+  children: PropTypes.node.isRequired,
+  type: PropTypes.oneOf(['bottomsheet', 'centered', 'fullscreen']),
```

Comment:
I missed this somehow. It should be `variant` basis the convention we follow across
```suggestion
  variant: PropTypes.oneOf(['bottomsheet', 'centered', 'fullscreen']),
```